lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php

```diff
@@ -32,8 +33,23 @@ class LazySchemaDiffProvider implements SchemaDiffProviderInterface
     /** @var SchemaDiffProviderInterface */
     private $originalSchemaManipulator;
 
-    public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaDiffProviderInterface $originalSchemaManipulator)
+    /** @var  Configuration */
+    private $proxyConfig;
+
+    public function __construct(LazyLoadingValueHolderFactory $proxyFactory=null, SchemaDiffProviderInterface $originalSchemaManipulator)
```

Comment:
Please don't allow the proxy factory to be null. The `LazySchemaDiffProvider` is a new class, and there is no need to provide defaults in this way. If you need to provide sane defaults, add a `public static function fromDefaultProxyFacyoryConfiguration()`, for example